applicationset/services/scm_provider/gitea.go

```diff
@@ -35,7 +35,7 @@ func NewGiteaProvider(ctx context.Context, owner, token, url string, allBranches
 	}
 	client, err := gitea.NewClient(url, gitea.SetToken(token), gitea.SetHTTPClient(httpClient))
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error creating a new git client: %w", err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("error creating a new gitea client: %w", err)
```
Just noticed this is specific to gitea.